contracts/RMRK/extension/erc20Holder/ERC20Holder.sol

```diff
@@ -19,32 +19,42 @@ abstract contract ERC20Holder is IERC20Holder {
         address erc20Contract,
         uint256 tokenId
     ) external view returns (uint256) {
-        return balances[tokenId][erc20Contract];
+        return _balances[tokenId][erc20Contract];
     }
 
     /**
-     * @inheritdoc IERC20Holder
+     * @notice Transfer ERC-20 tokens to a specific token
+     * @dev The ERC-20 contract must have approved this contract to transfer the ERC-20 tokens
```

Comment:
I don't think the approval is needed, since the tokens are owned by this smart contract.

```suggestion

```

Reply:
On the same not of the previous comment. The contract will pull them from user balances, so it needs to be approved.